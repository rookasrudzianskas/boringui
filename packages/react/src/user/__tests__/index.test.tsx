import React from "react";
import {mount, render} from "enzyme";

import User from "../index";

describe(" User", () => {
  it("should render correctly", () => {
    const wrapper = mount(<User name="Junior" />);

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should support image and text", () => {
    const wrapper = render(
      <div>
        <User name="Junior" text="Junior" />
        <User name="Junior Garcia" src="https://avatars.githubusercontent.com/u/30373425?v=4" />
      </div>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render description correctly", () => {
    const wrapper = mount(<User name="Junior">description</User>);

    expect(wrapper.text().toLowerCase()).toContain("description");
  });

  it("should render link on user.link", () => {
    const wrapper = mount(
      <User name="Junior">
        <User.Link href="https://twitter.com/jrgarciadev">twitter</User.Link>
      </User>,
    );
    const link = wrapper.find("a");

    expect(link.length).not.toBe(0);
  });

  it("should pass alt attribute", () => {
    const wrapper = mount(
      <User
        altText="Junior"
        name="Junior"
        src="https://avatars.githubusercontent.com/u/30373425?v=4"
      />,
    );

    expect(wrapper.find("img").prop("alt")).toEqual("Junior");
  });
});
