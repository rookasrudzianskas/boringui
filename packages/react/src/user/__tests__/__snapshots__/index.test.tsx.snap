// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` User should support image and text 1`] = `
LoadedCheerio {
  "0": <div>
    <div
      class="nextui-c-ijHtNE nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-ijHtNE-iPJLV-css"
    >
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar nextui-user-avatar"
        data-state="ready"
      >
        <span
          class="nextui-avatar-bg"
        />
        <span
          class="nextui-avatar-text"
        >
          Jun
        </span>
      </span>
      <div
        class="nextui-c-kNKVvs nextui-user-info"
      >
        <span
          class="nextui-c-hEcroZ nextui-user-name"
        >
          Junior
        </span>
        <span
          class="nextui-c-eCZQvV nextui-user-desc"
        />
      </div>
    </div>
    <div
      class="nextui-c-ijHtNE nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-ijHtNE-iPJLV-css"
    >
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-user-avatar"
        data-state="loading"
      >
        <span
          class="nextui-avatar-bg"
        />
        <img
          class="nextui-avatar-img nextui-avatar--loading"
          data-state="loading"
          src="https://avatars.githubusercontent.com/u/30373425?v=4"
        />
      </span>
      <div
        class="nextui-c-kNKVvs nextui-user-info"
      >
        <span
          class="nextui-c-hEcroZ nextui-user-name"
        >
          Junior Garcia
        </span>
        <span
          class="nextui-c-eCZQvV nextui-user-desc"
        />
      </div>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
